File: core/src/main/java/forpleuvoir/dhwu1a/core/message/messagesender/FriendMessageSender.java
package forpleuvoir.dhwu1a.core.message.messagesender;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.base.MessageSender;
import forpleuvoir.dhwu1a.core.user.Friend;

import static forpleuvoir.dhwu1a.core.common.ApiKey.NICKNAME;
import static forpleuvoir.dhwu1a.core.common.ApiKey.REMARK;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.messagesender
 * <p>#class_name FriendMessageSender
 * <p>#create_time 2021/7/1 21:19
 */
public class FriendMessageSender extends MessageSender {

    @SerializedName(NICKNAME)
    public String nickname;
    @SerializedName(REMARK)
    public String remark;

    protected FriendMessageSender(Long id) {
        super(id);
    }

    @Override
    public Friend getUser() {
        return bot.getFriend(this.id);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/messagesender/FriendMessageSender.kt
package forpleuvoir.dhwu1a.core.message.messagesender

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.NICKNAME
import forpleuvoir.dhwu1a.core.common.REMARK
import forpleuvoir.dhwu1a.core.message.base.MessageSender
import forpleuvoir.dhwu1a.core.user.Friend

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.messagesender
 *
 * #class_name FriendMessageSender
 *
 * #create_time 2021/7/1 21:19
 */
open class FriendMessageSender protected constructor(id: Long) : MessageSender(id) {
    @SerializedName(NICKNAME)
    var nickname: String? = null

    @SerializedName(REMARK)
    var remark: String? = null
    override val user: Friend
        get() = bot.getFriend(this.id)!!
}